File: kmp/nostr-sdk-kmp/examples/event_builder.kt
package org.nostrdevkit.examples

import org.nostrdevkit.sdk.*

fun eventBuilderExample(): List<String> {
    val keys = Keys.generate()
    val kind = Kind.fromStd(KindStandard.TEXT_NOTE)

    val textNote = EventBuilder(kind, "Note from rust-nostr Kotlin bindings").finalize(keys)
    val customBuilder = EventBuilder(Kind(1234u), "My custom content")
    val customEvent = customBuilder.finalize(keys)
    val powEvent = customBuilder.finalizeUnsigned(keys.publicKey())
        .mine(SingleThreadPow(), 8u)
        .sign(keys)
    val unsignedEvent = customBuilder.finalizeUnsigned(keys.publicKey())

    return listOf(
        textNote.asJson(),
        customEvent.asJson(),
        powEvent.asJson(),
        unsignedEvent.asJson(),
    )
}

File: js/web/examples/event_builder.mjs
import {
  EventBuilder,
  Keys,
  Kind,
  KindStandard,
  SingleThreadPow,
} from "../dist/index.js";

export function eventBuilderExample() {
  const keys = Keys.generate();
  const kind = Kind.fromStd(KindStandard.TextNote);
  const textNote = new EventBuilder(
    kind,
    "Note from rust-nostr Web bindings",
  ).finalize(keys);

  const customBuilder = new EventBuilder(new Kind(1234), "My custom content");
  const customEvent = customBuilder.finalize(keys);
  const powEvent = customBuilder
    .finalizeUnsigned(keys.publicKey())
    .mine(new SingleThreadPow(), 8)
    .sign(keys);
  const unsignedEvent = customBuilder.finalizeUnsigned(keys.publicKey());

  return [
    textNote.asJson(),
    customEvent.asJson(),
    powEvent.asJson(),
    unsignedEvent.asJson(),
  ];
}
